### PR TITLE
chore(react toolbar): add deprecation warning to Toolbar/ToolbarSearch

### LINKED ISSUE #3376: AVT 1 - React Toolbar has DAP violations for missing labels.
## Environment
macOS Mojave version 10.14.5
Chrome Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
DAP - July 2019 Ruleset

## Detailed Description
Run DAP on the Toolbar component and the following violations are present:
Screen Shot 2019-07-11 at 5 01 45 PM

### LINKED ISSUE #3381: AVT 3 - React Toolbar needs improvements for VO  keyboard navigation on iOS and macOS
1. Start VO on macOS
2. Tab to the Toolbar, both Search and Filter icons have focus. VO announces, “Search. You are currently on a text field. To enter this field, type.” (see screenshot)
Screen Shot 2019-07-11 at 5 07 47 PM **Expected Result:** The focus should only be on the Search icon and advise the user that it is a button.  The user should not be able to type anything yet. If the user does type a word it is saved in the search text field but the user can not use the Space or Enter key to submit the search and VO is silent.  

3. The only way to exit the search field is to Tab which takes the user to the Search button icon and VO announces that it is a button. (see screenshot)
Screen Shot 2019-07-11 at 5 08 23 PM**Expected Result:** The purpose of the button should be announced. 

4. Press the Space key to open the search and VO announces, “Search. You are currently on a text field. To enter this field, type.” Note: The text that was entered in step 2 is in the text field and the X icon that displays cannot get focus to clear the text field.
**Expected Result:** Text should not be present yet and the X icon should clear the text field.

5. Press Tab again and the focus moves back to the Search button and the Search field remains open. 
**Expected Result:** After the search, Tab should move to the X to Clear the search and then Tab again should move to the Filter button.

5. Press Tab again and the focus moves to the filter icon after the search icon and VO announces, “Menu pop-up button.  You are currently on a pop-up button. To select this pop-up button press Control-Option-Space.”
**Expected Result:** The purpose of the pop-up menu should be announced. 

6. Press the Space key and the Filter selections display. VO announces that you are on a menu that displays a list of  4 items and to move press Control-Option-Space. Moving to the first item on the list it is announced as 1 of 5, 2 of 5, 3 of 5, and 4 of 5. 
**Expected Result:** VO should announce that this is a list of 3 items… and moving to each item VO should announce the option 1 of 3 and the state, option 2 of 3 and the state, and option 3 of 3 and the state.

7. Press Esc Close the menu and then press Tab and VO announces, “Menu pop-up button.  You are currently on a pop-up button. To select this pop-up button press Control-Option-Space.”
**Expected Result:** The purpose of the pop-up menu should be announced.

8. Press the Space key and the Menu selections display. VO announces that you are on a menu that displays a list of  5 items and to move press Control-Option-Space. Moving to the first item on the list the Refresh Table text is announced as 1 of 6 and the remaining items are announced 2 of 6, 3 of 6, etc. 
**Expected Result:** VO should announce that this is a list of 3 items… and moving to each item should be Refresh table 1 of 3, Short Radio button 2 of 3 and the state, Tall radio button 3 of 3 and the state. Also, note that the horizontal rule does not need to have focus and it should not be counted as an item. 

9. Press Esc to close the pop-up menu.

WCAG 2.0 Requirements:
[Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)
[Meaningful Sequence: Understanding SC 1.3.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html)

Similar issues on iOS with VO when using gestures. iOS has addition focus issues when navigating in and out of pop-up menus.

### LINKED ISSUE #3382: AVT 2 - React Toolbar Keyboard only navigation and color contrast issues.
## Environment
macOS Mojave version 10.14.5
Chrome Version 75.0.3770.100 (Official Build) (64-bit)
Carbon v10 - React
Color Contrast Analyzer

## Detailed Description
1. Tab to the Seach icon
2. Press Space to open the Search text input field
3. Enter text and the X icon displays
4. Tab and the Search icon gets focus. **Expected Result:**  KB navigation to the X icon to clear the text. 
5. Press Space on the Search icon and the Search Text box closes, but the X icon is still present. (see screenshot):
Screen Shot 2019-07-12 at 11 17 11 AM

Run Color Contrast Analyzer on the X icon and Search placeholder text and they both fail WCAG color contrast requirements (see screenshots):
Screen Shot 2019-07-12 at 11 26 09 AM
Screen Shot 2019-07-12 at 11 30 12 AM

### PR DESCRIPTION
**Pull request by dakahn**
Closes #3381 
Closes #3382 
Closes #3376

Seems like after asking around the [Toolbar/ToolbarSearch](http://react.carbondesignsystem.com/?path=/story/toolbar--default) component wasn't supposed to have made it through to v10. Adding a deprecation warning now.

### CODE REVIEW

**Review by joshblack — CHANGES_REQUESTED** [review]
Ah, forgot to tell you about the `__DEV__` block pattern, my bad 🤦‍♂️ Also forgot about how we called warning in the past lol. Here is a quick link:

https://github.com/carbon-design-system/carbon-components-react/compare/v6.114.0...v7.0.0#diff-6652cf074cd01cfc7b6dd43982583869

tl;dr

```js
let didWarnAboutDeprecation = false;

// ...
constructor() {
  if (__DEV__) {
    warning(
      didWarnAboutDeprecation,
      'The `ComponentName` component has been deprecated and will be removed in the next major release of `carbon-components-react`'
    );
    didWarnAboutDeprecation = true;
  }
}
```

File: packages/react/src/components/ToolbarSearch/ToolbarSearch.js [hunk 1]
```diff
@@ -64,16 +64,14 @@ export default class ToolbarSearch extends Component {
...
   };
 
-  constructor(props) {
-    super(props);
-
-    if (didWarnAboutDeprecation === false) {
-      didWarnAboutDeprecation = true;
-
+  constructor() {
+    super();
```
Comment (asudoh) [inline packages/react/src/components/ToolbarSearch/ToolbarSearch.js]: Does `super()` need `props`?

Comment (jnm2377) [reply packages/react/src/components/ToolbarSearch/ToolbarSearch.js]: Yes, I'm not sure if it needs super in this case. I didn't see it in the examples, but I could be wrong.

Comment (dakahn) [reply packages/react/src/components/ToolbarSearch/ToolbarSearch.js]: Pretty sure its required if you're es6 class extends another class (or if you ref this.props in the constructor). I'll add it 👍

**Review by jnm2377 — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @dakahn!

**Comment by tay1orjones** [comment]
I'm testing out the 10.7 rc
The console error deprecation warning I see from this change doesn't contain the component name, it's quite literally `ComponentName`. Quite confusing 😅 
```
console.error node_modules/carbon-components-react/node_modules/warning/warning.js:51
      Warning: The `ComponentName` component has been deprecated and will be removed in the next major release of `carbon-components-react`
```

**Comment by joshblack** [comment]
Haha, wow, sorry we missed this 🤦‍♂ 

cc @dakahn mind making a quick PR to interpolate the component name? Sorry we missed this in review!

**Comment by joshblack** [comment]
@tay1orjones just curious, what are you all using the standalone toolbar for?

**Comment by tay1orjones** [comment]
@joshblack We use it and the ToolbarItem/Title/Option in the headers for our [Card](https://github.com/IBM/carbon-addons-iot-react/blob/f863eda94f7904ded9dc5b55b3704ce2188abd15/src/components/Card/Card.jsx#L5-L8) component which are placed in our [dashboard](https://carbon-addons-iot-react.com/?path=/story/watson-iot-experimental-dashboard--basic).
